public/src/managers/CourseManager.ts

```diff
@@ -353,11 +345,11 @@ export class CourseManager {
         let wantGroupLinks = false;
         if (student) {
             submissions =
-                await this.courseProvider.getAllLabInfos(courseLabLink.course.getId(), student.getId());
+                await this.courseProvider.getLabsForStudent(courseLabLink.course.getId(), student.getId());
```

Comment:
We should unify naming. That is, we should not refer to any `Lab`, and instead use `Assignment`

I'm also not clear about these legacy link objects, e.g. `courseLabLink` and `wantGroupLink`... I think what is being referred to is that they are the associations between two other tables to resolve a many-to-many relationship.

Reply:
@0xf8f8ff Did you see this?

Reply:
Yeah, I agree we should replace all "labs" with either "assignment" or "submission" for consistency.
The legacy link objects can probably be replaced by the protobuf defined types more easily now with the last TS changes. I will try to replace them in a separate PR.

Reply:
Should we also change LabRequest, isGroupLab and similar examples?

Reply:
Yeah probably. But it's okay to do it as a separate PR, or postpone it to be sure that we get time to implement the important stuff. Let me know, and I can merge this PR, and we continue with other stuff.